Tests/UnitTests/Networking/HTTPClientTests.swift

```diff
@@ -637,6 +637,37 @@ class HTTPClientTests: XCTestCase {
         expect(completionCalled.value).toEventually(equal(true), timeout: .seconds(1))
     }
 
+    func testGetsResponseFromETagManagerWhenStatusCodeIsNotModified() {
+        let path: HTTPRequest.Path = .mockPath
+
+        let mockedCachedResponse: [String: String] = [
+            "test": "data"
+        ]
+
+        let response: Atomic<(code: HTTPStatusCode, result: Result<HTTPResponse.Body, Error>)?> = .init(nil)
+
+        self.eTagManager.shouldReturnResultFromBackend = false
+        self.eTagManager.stubbedHTTPResultFromCacheOrBackendResult = .init(
+            statusCode: .notModified,
+            jsonObject: mockedCachedResponse
+        )
+
+        stub(condition: isPath(path)) { _ in
+            return .init(data: Data(),
+                         statusCode: .notModified,
+                         headers: nil)
+        }
+
+        self.client.perform(.init(method: .get, path: path), authHeaders: [:]) {
+            response.value = ($0, $1)
+        }
+
+        expect(response.value).toEventuallyNot(beNil(), timeout: .seconds(1))
+
+        expect(response.value?.code) == .notModified
```

Comment:
I think we should also make `HTTPClient` return `HTTPStatusCode.success` in this case, since users of this class don't care if the data was cached. Thoughts?

Reply:
Oh never mind, I see `ETagManager` already stores the original response code, I'll change the test